src/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -265,6 +257,7 @@ internal int ReadCore(Span<byte> buffer)
 
             while (true)
             {
+                Debug.Assert(_inflater != null);
```

Comment:
Move this out of the while, we shouldn't assert on every iteration as `_inflater` is not changing.